### DISCUSSION TITLE
CSS-compatible style syntax

### DISCUSSION THREAD

**Discussion by asbjornu** [plantuml/plantuml#789] [Ideas]
As suggested in https://github.com/plantuml/plantuml/issues/218#issuecomment-974157523, I'd like to make the case for the new [`style` feature](http://alphadoc.plantuml.com/doc/dokuwiki/en/style-evolution) that is in development to adopt a syntax that is compatible with CSS. Being compatible with CSS would make the syntax easier to read and write because many developers already know CSS.

The change wouldn't mean much for the proposed syntax, it would mostly affect comments and introducing a colon between properties and values. It would change the proposed syntax from:

```
' scope to sequenceDiagram elements
sequenceDiagram {
     ' scope to actor element types
     actor {
         FontColor Blue
     }
}
```

To:

```css
/* scope to sequenceDiagram elements */
sequenceDiagram {
     /* scope to actor element types */
     actor {
         FontColor: Blue
     }
}
```

**Comment by arnaudroques** [comment]
> Being compatible with CSS would make the syntax easier to read and write because many developers already know CSS.

Sure. Actually, both syntax are working. (see [here](https://www.plantuml.com/plantuml/uml/DOrB3i8m34JtFeKlWDYNgYeVSOyqZ22I6vVOMI3KkzC8UCkPqRpHF1Ilir1l_X8CP7XMv8XB2lSIPduJDvPjbc9QGlOM_RbgzhEAbeudgJXycvLMwlU_mvDEl1luA2c2Etv0n9H6vDkc_G00))

Comments like:

```
/* scope to sequenceDiagram elements */
```

are not possible yet but we are going to implement it right now (no need to open an issue for that).

Any feedback on this feature is greatly appreciate, so do not hesitate to continue the discussion :-)
